modules/selection/src/main/kotlin/com.r3.corda.lib.tokens.selection/memory/services/VaultWatcherService.kt

```diff
@@ -186,7 +294,7 @@ class VaultWatcherService(private val tokenObserver: TokenObserver,
             for (stateAndRef in stateAndRefs) {
                 val existingMark = __backingMap.putIfAbsent(stateAndRef, PLACE_HOLDER)
                 existingMark?.let {
-                    LOG.warn("Attempted to overwrite existing token ${stateAndRef.ref}, this suggests incorrect vault behaviours")
+                    LOG.warn("Attempted to overwrite existing token ${stateAndRef.ref}, this suggests a result set re-sync occurred")
```

Comment:
We should reconsider this warning.  Does something break when this condition occurs?  Should the function return or re-do the indexing (as it does now)?   Should it really be a `debug`?

Reply:
I've changed to a debug. It is just saying that a duplicate entry has been inserted. This would be because of a resync. Although in the code i do try to minimise the duplicate inserts by moving to the next unread entry (in all cases except one, where we have seen the whole list).